src/test/resources/hibernate.cfg.xml

```diff
@@ -0,0 +1,23 @@
+<?xml version = "1.0" encoding = "utf-8"?>
```

Comment:
Adding H2 was as simple as adding a separate hibernate.cfg.xml file in src/test/resources that uses the H2 dialect and driver class.